tests/Thumbnail/LiipImagineThumbnailTest.php

```diff
@@ -0,0 +1,118 @@
+
+        $image = $this->createMock(ImageInterface::class);
+        $image->expects($this->any())->method('getSize')->will($this->returnValue($size));
+
+        $adapter = $this->createMock(ImagineInterface::class);
+        $adapter->expects($this->any())->method('open')->will($this->returnValue($image));
+
+        $metadata = $this->createMock(MetadataBuilderInterface::class);
+
+        $provider = new ImageProvider('file', $filesystem, $cdn, $generator, $thumbnail, $allowedExtensions, $allowedMimeTypes, $adapter, $metadata);
+        $provider->setResizer($resizer);
+
+        return $provider;
+    }
+
+    public function testGenerate()
+    {
+        $cacheManager = $this->createMock(CacheManager::class);
+        $cacheManager->expects($this->any())->method('getBrowserPath')->will($this->returnValue('cache/media/default/0011/24/ASDASDAS.png'));
+
+        $thumbnail = new LiipImagineThumbnail($cacheManager);
+
+        $filesystem = new Filesystem(new InMemory(['myfile' => 'content']));
+        $referenceFile = new File('myfile', $filesystem);
+
+        $formats = [
+          'admin' => ['height' => 50, 'width' => 50, 'quality' => 100],
+          'mycontext_medium' => ['height' => 500, 'width' => 500, 'quality' => 100],
+          'anothercontext_large' => ['height' => 500, 'width' => 500, 'quality' => 100],
+        ];
+
+        $resizer = $this->createMock('Sonata\MediaBundle\Resizer\ResizerInterface');
+        $resizer->expects($this->any())->method('resize')->will($this->returnValue(true));
+
+        $provider = $this->createMock('Sonata\MediaBundle\Provider\MediaProviderInterface');
+        $provider->expects($this->any())->method('requireThumbnails')->will($this->returnValue(true));
+        $provider->expects($this->any())->method('getReferenceFile')->will($this->returnValue($referenceFile));
+        $provider->expects($this->any())->method('getFormats')->will($this->returnValue($formats));
+        $provider->expects($this->any())->method('getResizer')->will($this->returnValue($resizer));
+        $provider->expects($this->any())->method('generatePrivateUrl')->will($this->returnValue('/my/private/path'));
+        $provider->expects($this->any())->method('generatePublicUrl')->will($this->returnValue('/my/public/path'));
+        $provider->expects($this->any())->method('getFilesystem')->will($this->returnValue($filesystem));
+        $provider = $this->getProvider();
+        $media = new Media();
+        $media->setName('ASDASDAS.png');
+        $media->setProviderReference('ASDASDAS.png');
+        $media->setId(1023456);
+        $media->setContext('default');
+
+        $thumbnail->generate($provider, $media);
+        $this->assertSame('default/0011/24/ASDASDAS.png', $thumbnail->generatePublicUrl($provider, $media, MediaProviderInterface::FORMAT_ADMIN));
+        $this->assertSame('cache/media/default/0011/24/ASDASDAS.png', $thumbnail->generatePublicUrl($provider, $media, 'mycontext_medium'));
+    }
```

Comment:
Please add a test for the deprecated path

```php
<?php
/**
 * @group legacy
 * @expectedDeprecationMessage part of the expected message
 */
public function testWhateverYouWantToCallIt()
{
   // instantiate the object in the deprecated way and check it still works
}
```

Reply:
Sorry, I'm afraid I'm not capable of that. At this point I would like to hand over to someone else. Thanks all for your help!

Reply:
I'll try to do it this week-end then, can you please reopen?

Reply:
And can you please click the "allow maintainers to push" checkbox? Should be bottom right

Reply:
Ah wait nevermind, I can reopen.